sql3/planner/compileselect.go

```diff
@@ -83,12 +83,12 @@ func (p *ExecutionPlanner) compileSelectStatement(stmt *parser.SelectStatement,
 		for _, agg := range aggregates {
 			InspectExpression(agg, func(expr types.PlanExpression) bool {
 				switch ex := expr.(type) {
-				case *sumPlanExpression, *countPlanExpression, *countDistinctPlanExpression,
-					*avgPlanExpression, *minPlanExpression, *maxPlanExpression,
-					*percentilePlanExpression:
+				case types.Aggregable:
 					ch := ex.Children()
-					// first arg is always the ref
-					aggregateAndGroupByExprs = append(aggregateAndGroupByExprs, ch[0])
+					// first arg is always the ref, except for count(*)
+					if len(ch) > 0 {
+						aggregateAndGroupByExprs = append(aggregateAndGroupByExprs, ch[0])
+					}
 					return false
```

Comment:
I dont understand what is going on here.  Why is this returning false regardless of the conditional?  What purpose does this serve?  What is the use case of ch == 0?

Why not:
`if len(ex.Children()) > 0 {
  aggregateAndGroupByExprs = append(aggregateAndGroupByExprs, ex.Children()[0])
}`
Why create the new memory structure?  Is this a Go thing?

Reply:
The `return false` is for the closure that's passed to `InspectExpression`, it basically means stop inspecting. 

The `ch := ex.Children()` is just being explicit. The compiler will optimize it away.